main.tf

```diff
@@ -95,17 +95,17 @@ resource "azurerm_kubernetes_cluster" "main" {
     }
   }
   dynamic "azure_active_directory_role_based_access_control" {
-    for_each = var.role_based_access_control_enabled && var.rbac_aad_managed ? ["rbac"] : []
+    for_each = var.rbac_aad && var.rbac_aad_managed ? ["rbac"] : []
 
     content {
       admin_group_object_ids = var.rbac_aad_admin_group_object_ids
-      azure_rbac_enabled     = var.rbac_aad_azure_rbac_enabled
+      azure_rbac_enabled     = local.rbac_aad_azure_rbac_enabled
```

Comment:
What if we declare an aks resource with an `azure_active_directory_role_based_access_control` block, but omit it's `azure_rbac_enabled` argument? I mean, if we set `null` to this argument, is that a valid configuration?

Reply:
It is a valid configuration. In the `master` branch the variable `rbac_aad_azure_rbac_enabled` has already a default value of `null`. This is because [azure_rbac_enabled](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#azure_rbac_enabled) is optional.

This is a valid configuration and it means:
* Enable the Active Directory integration for authentication
* Do not use Active Director for RBAC

This is the equivalent of doing `az aks create  --enable-aad` **without** the parameter `--enable-azure-rbac`.

Reply:
We should avoid having the `local` and use directly the variable then ?